fix(middleware): Return remote_addr without port if it is available.

blardo opened the pull request.

ethanfrogers commented on middlewares/middlewares.go:
```diff
@@ -73,7 +74,11 @@ func getRemoteAddr(r *http.Request) string {
 	if len(address) > 0 {
 		return address
 	}
+	if splitIndex := strings.Index(r.RemoteAddr, ":"); splitIndex > 0 {
```
`-1` denotes the nonexistence of a substring in `strings.Index`. If for some reason `:` is the first character in the string, this will still return `:8907`. It's _probably_ okay though.

ethanfrogers commented on middlewares/middlewares.go:
```diff
@@ -73,7 +74,11 @@ func getRemoteAddr(r *http.Request) string {
 	if len(address) > 0 {
 		return address
 	}
+	if splitIndex := strings.Index(r.RemoteAddr, ":"); splitIndex > 0 {
```
`strings.Index` returns `-1` to denote nonexistence of a substring. This could still produce a bad result if (for some reason) `r.RemoteAddr` started with a `:`.

micahhausler replied:
You'll probably need to use `strings.LastIndex()` to account for ipv6 `[::]:1234`

ethanfrogers reviewed (approved):
LGTM. Just one comment about `strings.Index` that might be a non-issue.

micahhausler commented on middlewares/middlewares_test.go:
```diff
@@ -0,0 +1,51 @@
+
+	cases := []struct {
+		name      string
+		header    string
```
Since you're setting header every time, I don't think this test will ever catch the change you made. You can check this by running:

```bash
go test -coverprofile=coverage.out ./middlewares/
go tool cover -html=coverage.out
```

You could probably make `header` a `map[string]string` and set the `ipAddress` field to the `RemoteAddr` field on your request after you create it.

As @ethanfrogers pointed out, it might be good to add some test cases that include header or RemoteAddr values like `[::]:1234` (ipv6 localhost)

micahhausler reviewed (changes requested):
Great start! Just make sure you're testing all the branches on the function.

Commit:
fix(middleware): Return remote_addr without port if it is available.

feature(test): add first middleware test
feat(tests): Remove extra coverage runs to simplify travis output.

Add coverage to test run

refactor(getRemoteAddr): Refactor returns to use stripPort

micahhausler reviewed (approved):
👍 Great work!
